agent/lm_agent/reconciliation.py

```diff
@@ -144,55 +145,146 @@ async def filter_cluster_update_licenses(licenses_to_update: List) -> List:
     return filtered_licenses
 
 
+async def get_booking_sum_for_cluster(cluster_name: str) -> int:
+    """
+    Get booking sum for bookings in the specified cluster.
+    """
+    response = await backend_client.get("/lm/api/v1/booking/all")
+    bookings = response.json()
+
+    booking_sum = 0
+
+    for booking in bookings:
+        if booking["cluster_name"] == cluster_name:
+            booking_sum += booking["booked"]
+
+    return booking_sum
+
+
+async def get_booking_sum_for_other_clusters(cluster_name: str) -> int:
+    """
+    Get booking sum for bookings in clusters other than the specified cluster.
+    """
+    response = await backend_client.get("/lm/api/v1/booking/all")
+    bookings = response.json()
+
+    booking_sum = 0
+
+    for booking in bookings:
+        if booking["cluster_name"] != cluster_name:
+            booking_sum += booking["booked"]
+
+    return booking_sum
+
+
+async def create_or_update_reservation(reservation_data):
+    """
+    Create the reservation if it doesn't exist, otherwise update it.
+    If the reservation cannot be updated, delete it and create a new one.
+    """
+    reservation = await scontrol_show_reservation()
+
+    if reservation:
+        updated = await scontrol_update_reservation(reservation_data, "30:00")
+        if not updated:
+            deleted = await scontrol_delete_reservation()
+            if not deleted:
+                raise LicenseManagerReservationFailure("Could not update or delete reservation.")
```

Comment:
You could use the `require_condition()` helper here for some syntactical sugar.

Reply:
I updated it in the latest commit.